### DISCUSSION TITLE
Reduce notifications?

### DISCUSSION THREAD

**Discussion by kentcdodds** [johnlindquist/kit#621] [Ideas]
I feel like every day I start with a notification from Kit about an update:

![image](https://user-images.githubusercontent.com/1500684/143722375-ca8deeeb-d7a1-4d1f-b97b-16e68fd6b7b6.png)

It's just a lot. Could I just be notified if there's a problem updating or if I try to use Kit during an update or something?

**Comment by johnlindquist** [comment]
Which version are you on? The "problem updating" notification shouldn't even exist anymore

**Reply by johnlindquist** [reply]
And yes, there will be _much_ fewer updates in the near future. (Once a month, maybe?) I'm polishing small things for the main channel release, so builds are way more frequent this past week.

**Reply by kentcdodds** [reply]
I'm on version 1.3.0-beta.104

**Reply by johnlindquist** [reply]
Hrmm, that's failing to update then, that's why so many notifications show up. .104 may have been the tail-end of versions with updating issues.

Will you please post the kit.log to help me out?

This will copy them to the clipboard, then paste them in a reply:
```bash
cat ~/.kit/logs/kit.log  | pbcopy
```

## "Solution"
`Kit->Quit` then download the latest version from scriptkit.com and overwrite. That'll definitely get rid of the notifications and should have the updating under control.

**Reply by kentcdodds** [reply]
Why do I have to be the special case 😂😆😭

Here's [my kit.log](https://gist.github.com/kentcdodds/952da6c8164e6d669dc346342f244979)

**Reply by johnlindquist** [reply]
I imagine it happened to many people. You're just kind enough to report it 😅

**Reply by johnlindquist** [reply]
Thanks for the log. A search for: `Updating from 1.3.0-beta.104` shows that it kept downloading the latest update, but never successfully applied it, so it was stuck on .104. That's the painful part of update failures, the only solution is "download manually from scriptkit.com and overwrite".

## TMI Because I Can't Help Myself

I'm pretty sure the .106 update fixed it. You can see where I switch from electron's native `quitAndInstall` to "electron-updater"'s `quitAndInstall`:

https://github.com/johnlindquist/kitapp/compare//v1.3.0-beta.104...v1.3.0-beta.106#diff-0dd1092b99d3bb57c98957de5ec7d4f502b3470e49f3442d152980e40deb0a60R89

The docs are _really_ sparse, so it's been infuriating to debug, especially trying to wrangle the download/update events and version/channel comparisons:

https://www.electron.build/auto-update.html#module_electron-updater.AppUpdater+quitAndInstall

In your case, `quitAndInstall` was failing. I *think* manually quitting and restarting the app might have triggered the install, but we're probably too late now.

**Reply by kentcdodds** [reply]
Neato! I'm on the latest 115 version now. I'll keep you updated when funny things happen 😅 Thanks!
